components/director/internal/tenantmapping/model.go

```diff
@@ -0,0 +1,64 @@
+package tenantmapping
+
+import "github.com/kyma-incubator/compass/components/director/internal/model"
+
+// CredentialSrc wraps possible sources of credentials like token, username and certificate
+type CredentialSrc string
+
+// RefObjectType wraps possible options for the origin object represented by the credential
+type RefObjectType string
+
+func sysAuthToRefObjectType(sysAuth model.SystemAuth) RefObjectType {
+	if sysAuth.AppID != nil {
+		return Application
+	} else if sysAuth.RuntimeID != nil {
+		return Runtime
+	} else if sysAuth.IntegrationSystemID != nil {
+		return IntegrationSystem
+	}
+
+	return ""
+}
+
+const (
+	Application       RefObjectType = "Application"
```

Comment:
maybe rename to `XRefObjectType`, such as `ApplicationRefObjectType` etc. - it will make clear that these values are enum-like ones

Reply:
Well, I don't get the benefit from such change, could you elaborate a bit more?